### PR TITLE
Added Gaussian Discriminant Analysis model

### LINKED ISSUE #3191: [Feature Request]: Add Gaussian Discriminant Analysis model in Machine Learning
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Gaussian Discriminant Analysis (GDA) is a classification algorithm that assumes class-conditional densities follow Gaussian distributions. It estimates parameters such as mean and covariance matrix for each class from training data, allowing it to compute posterior probabilities of class membership for new inputs using Bayes' theorem. GDA is efficient with small datasets, interpretable due to its probabilistic nature, and applicable in scenarios where Gaussian distributions reasonably describe class distributions, such as in facial recognition, medical diagnostics, and text categorization tasks. However, its performance relies on the accuracy of the Gaussian assumption and the availability of sufficient training data for reliable covariance estimation.

### Use Case

-Medical Diagnostics
-Financial Fraud Detection

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by pavitraag**
## Description

Added Gaussian Discriminant Analysis in Machine Learning
Closes #3191 

## Type of PR

- [ ] Bug fix
- [X] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have followed the code style guidelines of this project.
- [X] I have checked for any existing open issues that my pull request may address.
- [X] I have ensured that my changes do not break any existing functionality.
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X] I have read the resources for guidance listed below.
- [X] I have followed security best practices in my code changes.

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: docs/Machine Learning/Gaussian-Discriminant-Analysis.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+sidebar_position: 1
+tags: [Gaussian Discriminant Analysis, GDA, machine learning, classification algorithm, data analysis, data science, probabilistic modeling, supervised learning, discriminative model, Gaussian distribution, feature modeling, pattern recognition, Gaussian Naive Bayes]
+description: In this tutorial, you will learn about Gaussian Discriminant Analysis (GDA), its importance, what GDA is, why learn GDA, how to use GDA, steps to start using GDA, and more.
+---
+
+### Introduction to Gaussian Discriminant Analysis
+Gaussian Discriminant Analysis (GDA) is a classical supervised learning algorithm used for classification tasks. It models the probability distributions of features for each class using Gaussian distributions. GDA aims to classify new data points based on the likelihood that they belong to a particular class, making it a powerful tool for probabilistic classification tasks.
+
+### What is Gaussian Discriminant Analysis?
+Gaussian Discriminant Analysis involves modeling the conditional probability distributions of features $X$ given each class \( y \):
```
Comment (Ajay-Dhangar) [inline docs/Machine Learning/Gaussian-Discriminant-Analysis.md:14]: replace `\( y \)` to `$y$`

File: docs/Machine Learning/Gaussian-Discriminant-Analysis.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+- **Single Class Model**Each class \( y \) is characterized by its own Gaussian distribution parameters:
```
Comment (Ajay-Dhangar) [inline docs/Machine Learning/Gaussian-Discriminant-Analysis.md:16]: replace `\( y \)` to `$ y $`

File: docs/Machine Learning/Gaussian-Discriminant-Analysis.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  
+  - **Mean**: Represents the average value or center of the distribution.
+  
+  - **Covariance**: Describes how the features of the data are correlated with each other.
+
+- **Decision Rule**: Classify new data points by choosing the class that maximizes the posterior probability \( P(y | X) \) using Bayes' theorem.
```
Comment (Ajay-Dhangar) [inline docs/Machine Learning/Gaussian-Discriminant-Analysis.md:22]: replace `\( P(y | X) \)` to `$P(y | x)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by pavitraag** [comment]
changes are done @Ajay-Dhangar

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**